src/segment_wifi_test.go

```diff
@@ -0,0 +1,194 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type wifiArgs struct {
+	commandOutput   string
+	commandError    error
+	hasCommand      bool
+	displayError    bool
+	segmentTemplate string
+}
+
+type netshStringArgs struct {
+	state          string
+	ssid           string
+	radioType      string
+	authentication string
+	channel        int
+	receiveRate    int
+	transmitRate   int
+	signal         int
+}
+
+func getNetshString(args *netshStringArgs) string {
+	const netshString string = `
+	There is 1 interface on the system:
+
+	Name                   : Wi-Fi
+	Description            : Intel(R) Wireless-AC 9560 160MHz
+	GUID                   : 6bb8def2-9af2-4bd4-8be2-6bd54e46bdc9
+	Physical address       : d4:3b:04:e6:10:40
+	State                  : %s
+	SSID                   : %s
+	BSSID                  : 5c:7d:7d:82:c5:73
+	Network type           : Infrastructure
+	Radio type             : %s
+	Authentication         : %s
+	Cipher                 : CCMP
+	Connection mode        : Profile
+	Channel                : %d
+	Receive rate (Mbps)    : %d
+	Transmit rate (Mbps)   : %d
+	Signal                 : %d%%
+	Profile                : ohsiggy
+
+	Hosted network status  : Not available`
+
+	return fmt.Sprintf(netshString, args.state, args.ssid, args.radioType, args.authentication, args.channel, args.receiveRate, args.transmitRate, args.signal)
+}
+
+func bootStrapEnvironment(args *wifiArgs) *wifi {
```

Comment:
We've stepped away from this in favor of using test cases, see [here](https://github.com/JanDeDobbeleer/oh-my-posh/blob/accf46420ff2924e863f5a76179abc92b2212535/src/segment_aws_test.go#L9) for example.

Reply:
I tried to see how to write these tests using test cases. But I really didn't like the structure of it. Two things I couldn't really grasp:

1. How do you handle mocking dependencies? 
2. How do you handle test cases that don't assert just the simple output of the function?

Too much logic and indirection in testing code makes tests much harder to maintain imo. This is your repository, so I'll check out the link you provided and do my best to restructure.

Reply:
I can always help out. Haven't found a downside as it only moves the bootstrap inside the for loop rather than have x functions. It also helps extending use-cases a lot faster.

Reply:
I will take a kick at it. I've pushed up the other changes you requested and suggested. This is the last piece.

Reply:
Done. And I see the light :) It's way cleaner.